src/Main.elm

```diff
@@ -1,10 +1,396 @@
 module Main exposing (main)
 
-import Data.Audience
-import Data.AudienceFolder
+import Browser
+import Data.Audience exposing (Audience)
+import Data.AudienceFolder exposing (AudienceFolder)
+import Data.AudienceTree exposing (AudienceTree)
 import Html exposing (Html)
+import Html.Attributes
+import Html.Events
+import Json.Decode exposing (Decoder)
+import Process
+import Task
 
 
-main : Html msg
+type Model
+    = Loading LoadingModel
+    | DecodingError
+    | Loaded LoadedModel
+
+
+type Msg
+    = LoadingMsg LoadingMsg
+    | LoadedMsg LoadedMsg
+
+
+{-| simulates a http loading
+-}
+wait : Msg -> Cmd Msg
+wait msg =
+    Task.perform (always msg) (Process.sleep 1000)
+
+
+listInData : Decoder a -> Decoder (List a)
+listInData decoder =
+    Json.Decode.field "data" (Json.Decode.list decoder)
```

Comment:
An opinion: I like it more if all these decoder field traversals are inside a single function, all at once, to keep it understandable and not have to chase the magic around various functions and modules -- so I'd keep this inline in the `Http.get` call or wherever the top-level decoder for `List Audience` has to live.